controllers/contentlibrary/contentlibraryitem/contentlibraryitem_controller.go

```diff
@@ -114,9 +114,18 @@ func (r *Reconciler) ReconcileDelete(clItem *imgregv1a1.ContentLibraryItem) erro
 		return nil
 	}
 
-	r.Metrics.DeleteMetrics(logger, vmiName, clItem.Namespace)
+	vmi := &vmopv1.VirtualMachineImage{}
+	namespace := clItem.Namespace
+	if err := r.Get(ctx, types.NamespacedName{Name: vmiName, Namespace: namespace}, vmi); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	r.Metrics.DeleteMetrics(logger, vmiName, namespace)
```

Comment:
Should this be moved inside the `if controllerutil.ContainsFinalizer(vmi, utils.VirtualMachineImageFinalizer)` blocker?

Reply:
I ended up putting it inside with moving the finalizer to item resources.